src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
         }
     });
 
+    /**
+     * Fetch fusion data from the backend API with following request parameters:
+     * - entrez gene ids
+     * - sample molecular identifiers
+     * - molecular profile ids (only when sample molecular identifiers are not provided)
+     * @type {MobxPromiseUnionType<StructuralVariant[]>}
+     */
+    readonly fusions = remoteData<StructuralVariant[]>({
+        await: () => [
+            this.genes,
+            this.selectedMolecularProfiles,
+            this.samples,
+            this.studyIdToStudy
+        ],
+        invoke: async () => {
+            let molecularProfiles: Array<string> = [];
+            let samples: SampleMolecularIdentifier[] = [];
```

Comment:
i don't think we need initliazation of samples up here (nor empty array assignment) because it's only used inside if statement and is assigned to result of map.

Reply:
same is true of molecularProfiles.

Reply:
yes I agree. I think you don't even need the variables because they're immediately used and not used again - just set params.X = ...